src/wwwroot/template-editor/adapterService.js

```diff
@@ -0,0 +1,246 @@
+          },
+          {
+            tagName: 'RSSFEED:DESCRIPTION',
+            tagReplace: '¿Quieres dar a conocer tu marca, atraer clientes y diferenciarte de tu competencia? El Marketing de Contenidos y este super eBook gratuito pueden ayudarte a lograrlo. ¡Enterate más!'
+          },
+          {
+            tagName: 'RSSITEM:TITLE',
+            tagReplace: 'Nuevo Ebook: Conquista a tu audiencia con contenidos irresistible'
+          },
+          {
+            tagName: 'RSSITEM:CONTENT',
+            tagReplace: '¿Quieres dar a conocer tu marca, atraer clientes y diferenciarte de tu competencia? El Marketing de Contenidos y este super eBook gratuito pueden ayudarte a lograrlo. ¡Enterate más! Hace tiempo que la publicidad  no tiene el impacto que solía tener. Las personas ya no creen en lo que las empresas dicen sobre sus propios productos. Y eso es obvio, ¿verdad?'
+          },
+          {
+            tagName: 'RSSITEM:URL',
+            tagReplace: 'http://www.google.com.ar/'
+          }
+        ];
+        rssTags.forEach(function(tagObj) {
+          html = html.replace(new RegExp('\\[\\[\\[' + tagObj.tagName + '\\]\\]\\]', 'g'), tagObj.tagReplace);
+        });
+        var response = {
+          success: true,
+          errorMessage: '',
+          subjectCampaign: '',
+          html: html
+        };
+        var fakePromise=$q.defer();
+        fakePromise.resolve(response);
+        return fakePromise.promise;
+      }
+
+      function campaignSaveChangesAsTemplate(params) {
+        /*
+          Return campaign state and convert/create in template at template folder
+        */
+        console.log("campaignSaveChangesAsTemplate");
+        var savedCampaignAsTemplate = {
+          Success: true,
+          ErrorMessage: '',
+          UrlToRedirect: 'http://www.google.com/',
+          IdTemplate: 1 //This is generated based on the templates already created in template folder or 1 if it's empty
+        }
+        return savedCampaignAsTemplate;
+      }
+
+      function campaignSaveChanges(params) {
+        /*
+          Return save success flag and url to redirect when edit step is finished
+        */
+        console.log("campaignSaveChanges");
+        var savedCampaign = {
+          data: {
+            Success: true,
+            ErrorMessage: '',
+            UrlToRedirect: 'http://www.google.com/'
+          }
+        }
+        var fakePromise = $q.defer();
+        fakePromise.resolve(savedCampaign);
+        return fakePromise.promise;
+      }
+
+      function getCampaign(id, useEditorAsTemplate) {
+        /*
+          Get json with campaign configuration
+        */
+        console.log("getCampaign");
+        var campaign = {
+          id: id,
+          type: 'campaign',
+          name: 'Campaign ' + id,
+          attributes: {},
+          innerHTML: '',
+          children: []
+        };
+        return $http.get("").then(function(){
+          return {data: campaign};
```

Comment:
I cannot understand it. Are you doing an HTTP GET?

Reply:
Why not simply do 
```js
var campaign = {
  id: id,
  type: 'campaign',
  name: 'Campaign ' + id,
  attributes: {},
  innerHTML: '',
  children: []
};
$q.resolve({ data: campaign });
```

Reply:
I will change all references to $q.defer to this approach, is more cleanest, thanks andres.